libs/sdk-ui-charts/src/highcharts/chartTypes/_chartOptions/chartOptionsBuilder.ts

```diff
@@ -410,14 +411,15 @@ export function getChartOptions(
         ? getCategoriesForTwoAttributes(viewByAttribute, viewByParentAttribute, emptyHeaderTitle)
         : getCategories(type, measureGroup, viewByAttribute, stackByAttribute, emptyHeaderTitle);
 
-    // When custom sorting is enabled and is Pie|Donut|Funnel chart, need to skip this, so the sort specified by the user does not get override.
-    const isAutoSortableChart = isOneOfTypes(type, [
-        VisualizationTypes.DONUT,
-        VisualizationTypes.PIE,
-        VisualizationTypes.FUNNEL,
-    ]);
+    // When custom sorting is enabled and is Pie|Donut|Funnel|Pyramid_with_attribute chart,
+    // need to skip this, so the sort specified by the user does not get overriden.
+    const isPyramidWithViewByAttribute = isPyramid(type) && Boolean(viewByAttribute);
+    const isAutoSortableChart =
+        isOneOfTypes(type, [VisualizationTypes.DONUT, VisualizationTypes.PIE, VisualizationTypes.FUNNEL]) ||
+        isPyramidWithViewByAttribute;
```

Comment:
I suggest one small improvement: move the code above to `isAutoSortableChart` function and just call it in `if` statement to make this bit more readable.